src/utils/storage.ts

```diff
@@ -0,0 +1,49 @@
+export const getUsername = (address: string): string | null => {
+  return localStorage.getItem(address);
+};
+
+export const setUsername = (address: string, username: string): void => {
+  localStorage.setItem(address, username);
+};
+
+export const getUserPassphrase = (): string | null => {
+  return localStorage.getItem('passphrase');
+};
+
+export const setUserPassphrase = (passphrase: string): void => {
+  localStorage.setItem('passphrase', passphrase);
+};
+
+export const getChannelPassphrase = (id: string): string | null => {
+  return localStorage.getItem('channelId-' + id);
+};
+
+export const setChannelPassphrase = (id: string, passphrase: string): void => {
+  localStorage.setItem('channelId-' + id, passphrase);
+};
+
+export const removeChannelPassphrase = (id: string): void => {
+  localStorage.removeItem('channelId-' + id);
+};
+
+export const getChannelAlias = (id: string): string | null => {
+  return localStorage.getItem('channelAlias-' + id);
+};
+
+export const setChannelAlias = (id: string, alias: string): void => {
+  localStorage.setItem('channelAlias-' + id, alias);
+};
+
+export const getUniqueChannels = (): string[] => {
+  const channels = Object.keys(localStorage).filter(obj => obj.match(/channelId/));
```

Comment:
This operation can get quite expensive if the localstorage is large. It would be cheaper if you would store all of the channels in an array that you store either as a buffer or JSON and do key-based look-ups and filtering on it.

Ideally this wouldn't even be stored in the localstorage until the user closes the page so that you can keep everything in-memory and fast.

Reply:
Will be resolved in `0.10.0`